JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt

```diff
@@ -104,25 +108,84 @@ fun ArticleScreen(
 }
 
 /**
- * Stateless Article Screen that displays a single post.
+ * Stateless Article Screen that displays a single post adapting the UI to different screen sizes.
  *
  * @param post (state) item to display
+ * @param showNavRail (state) whether the Drawer or NavigationRail needs to be shown
  * @param onBack (event) request navigate back
  * @param isFavorite (state) is this item currently a favorite
  * @param onToggleFavorite (event) request that this post toggle it's favorite state
  */
 @Composable
-fun ArticleScreen(
+private fun ArticleScreen(
     post: Post,
+    showNavRail: Boolean,
     onBack: () -> Unit,
     isFavorite: Boolean,
     onToggleFavorite: () -> Unit
 ) {
+    var showUnimplementedActionDialog by rememberSaveable { mutableStateOf(false) }
+    if (showUnimplementedActionDialog) {
+        FunctionalityNotAvailablePopup { showUnimplementedActionDialog = false }
+    }
 
-    var showDialog by rememberSaveable { mutableStateOf(false) }
-    if (showDialog) {
-        FunctionalityNotAvailablePopup { showDialog = false }
+    Row(Modifier.fillMaxSize()) {
+        val context = LocalContext.current
+        if (showNavRail) {
+            InterestsNavRail(
+                onBack = onBack,
+                onUnimplementedAction = { showUnimplementedActionDialog = true },
+                isFavorite = isFavorite,
+                onToggleFavorite = onToggleFavorite,
+                onSharePost = { sharePost(post, context) }
+            )
+        }
+        ArticleScreenContent(
+            post = post,
+            // Allow opening the Drawer if the NavRail is not on the screen
+            navigationIconContent = if (!showNavRail) {
+                {
+                    IconButton(onClick = onBack) {
```

Comment:
For some reason this onClick is only working for the IconButton, but not the Icon inside. You have to click outside of the arrow for it to work!

It's strange, Icon is setting semantics but not clickable anywhere.

Reply:
It's working locally™ for me. Does it happen with other emulators?